app/pages/corrections/CorrectionCountryPage.scala

```diff
@@ -17,17 +17,26 @@
 package pages.corrections
 
 import controllers.routes
-import models.{Country, UserAnswers}
+import models.{Country, Index, Mode, UserAnswers}
+import pages.PageConstants.{correctionToCountry, corrections}
 import pages.QuestionPage
 import play.api.libs.json.JsPath
 import play.api.mvc.Call
 
-case object CorrectionCountryPage extends QuestionPage[Country] {
+case class CorrectionCountryPage(periodIndex: Index, countryIndex: Index) extends QuestionPage[Country] {
 
-  override def path: JsPath = JsPath \ toString
+  override def path: JsPath = JsPath \ corrections \ periodIndex.position \ correctionToCountry \ countryIndex.position \ toString
 
   override def toString: String = "correctionCountry"
 
-  override def navigateInNormalMode(answers: UserAnswers): Call =
-    routes.IndexController.onPageLoad()
+   def navigate(mode: Mode, answers: UserAnswers, countriesFromNi: Seq[Country], countriesFromEU: Seq[Country]): Call = {
+    answers.get(CorrectionCountryPage(periodIndex, countryIndex)) match {
+      case Some(country) =>
+        if(countriesFromNi.contains(country) || countriesFromEU.contains(country)){
```

Comment:
We could combine the two lists where they are passed in to this method and only have one list to check if country exists... one to think about

Reply:
done now